browser/ai_chat/ai_chat_brave_search_throttle_browsertest.cc

```diff
@@ -0,0 +1,238 @@
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "content/public/test/test_navigation_observer.h"
+#include "content/public/test/test_utils.h"
+#include "net/base/net_errors.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace {
+
+constexpr char kBraveSearchHost[] = "search.brave.com";
+constexpr char kLeoPath[] = "/leo";
+constexpr char kOpenLeoButtonValidPath[] = "/open_leo_button_valid.html";
+constexpr char kOpenLeoButtonInvalidPath[] = "/open_leo_button_invalid.html";
+
+}  // namespace
+
+class AIChatBraveSearchThrottleBrowserTest : public InProcessBrowserTest {
+ public:
+  AIChatBraveSearchThrottleBrowserTest()
+      : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {}
+
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+    content::SetupCrossSiteRedirector(&https_server_);
+
+    base::FilePath test_data_dir =
+        base::PathService::CheckedGet(brave::DIR_TEST_DATA);
+    test_data_dir = test_data_dir.AppendASCII("leo");
+    https_server_.ServeFilesFromDirectory(test_data_dir);
+    ASSERT_TRUE(https_server_.Start());
+
+    permissions::PermissionRequestManager* manager =
+        permissions::PermissionRequestManager::FromWebContents(
+            GetActiveWebContents());
+    prompt_factory_ =
+        std::make_unique<permissions::MockPermissionPromptFactory>(manager);
+  }
+
+  void TearDownOnMainThread() override {
+    prompt_factory_.reset();
+    InProcessBrowserTest::TearDownOnMainThread();
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    InProcessBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  content::WebContents* GetActiveWebContents() {
+    return browser()->tab_strip_model()->GetActiveWebContents();
+  }
+
+  void ClickOpenLeoButton() {
+    // Modify the href to have test server port.
+    std::string script = R"(
+      var link = document.getElementById('continue-with-leo')
+      var url = new URL(link.href)
+      url.port = '$1'
+      link.href = url.href
+      link.click()
+    )";
+    std::string port = base::NumberToString(https_server_.port());
+
+    ASSERT_TRUE(content::ExecJs(
+        GetActiveWebContents()->GetPrimaryMainFrame(),
+        base::ReplaceStringPlaceholders(script, {port}, nullptr)));
```

Comment:
Maybe this could be just:

```cxx
ASSERT_TRUE(content::ExecJs(
  GetActiveWebContents()->GetPrimaryMainFrame(), content::JsReplace(R"(
      var link = document.getElementById('continue-with-leo')
      var url = new URL(link.href)
      url.port = '$1'
      link.href = url.href
      link.click()
    )", https_server_.port()
)));
```

Reply:
https://github.com/brave/brave-core/pull/26082/commits/c82227cd44b8e34008aba9e0eb8d3fe607a2152c